core/src/text/fontContext.cpp

```diff
@@ -40,35 +45,54 @@ void FontContext::unlock() {
     m_contextMutex.unlock();
 }
 
-bool FontContext::addFont(const std::string& _family, const std::string& _weight, const std::string& _style) {
+FontID FontContext::addFont(const std::string& _family, const std::string& _weight,
+                            const std::string& _style, bool _tryFallback) {
+
+    unsigned int dataSize = 0;
+    unsigned char* data = nullptr;
+    int font = FONS_INVALID;
 
     std::string fontKey = _family + "_" + _weight + "_" + _style;
-    if (m_fonts.find(fontKey) != m_fonts.end()) {
-        return true;
-    }
 
-    unsigned int dataSize;
-    unsigned char* data = nullptr;
+    auto it = m_fonts.find(fontKey);
+    if (it != m_fonts.end()) {
+        if (it->second < 0) {
+            goto fallback;
+        }
+        return it->second;
+    }
 
-    // Assuming bundled ttf file follows this convention
-    auto bundledFontPath = "fonts/" + _family + "-" + _weight + _style + ".ttf";
-    if ( !(data = bytesFromResource(bundledFontPath.c_str(), &dataSize))) {
-        const std::string sysFontPath = systemFontPath(_family, _weight, _style);
-        if ( !(data = bytesFromFileSystem(sysFontPath.c_str(), &dataSize)) ) {
-            return false;
+    {
+        // Assuming bundled ttf file follows this convention
+        auto bundledFontPath = "fonts/" + _family + "-" + _weight + _style + ".ttf";
+        if ( !(data = bytesFromResource(bundledFontPath.c_str(), &dataSize))) {
+            const std::string sysFontPath = systemFontPath(_family, _weight, _style);
+            if ( !(data = bytesFromFileSystem(sysFontPath.c_str(), &dataSize)) ) {
+
+                LOGE("Could not load font file %s", fontKey.c_str());
+                m_fonts.emplace(std::move(fontKey), INVALID_FONT);
+                goto fallback;
+            }
         }
     }
 
-    int font = fonsAddFont(m_fsContext, fontKey.c_str(), data, dataSize);
+    font = fonsAddFont(m_fsContext, fontKey.c_str(), data, dataSize);
 
     if (font == FONS_INVALID) {
-        logMsg("[FontContext] Error loading font %s\n", fontKey.c_str());
-        return false;
+        LOGE("Could not load font %s", fontKey.c_str());
+        m_fonts.emplace(std::move(fontKey), INVALID_FONT);
+        goto fallback;
     }
 
     m_fonts.emplace(std::move(fontKey), font);
 
-    return true;
+    return font;
+
+fallback:
+    if (_tryFallback && m_fonts.size() > 0) {
```

Comment:
Are we not always certain that m_fonts will have at least one font.. (We are bundling one font always for debugging texts).